### PR TITLE
feat(gateway): compare permission checks to permit

### PR DESCRIPTION
**Pull request by jonaro00**
## Description of change

Compare project access to Permit project access.
Use Permit for project list.
Remove project list pagination.

## How has this been tested? (if applicable)

Staging

### CODE REVIEW

File: gateway/src/auth.rs [hunk 1]
```diff
@@ -32,18 +32,15 @@ where
 {
     type Rejection = Error;
 
-    async fn from_request_parts(parts: &mut Parts, state: &S) -> Result<Self, Self::Rejection> {
+    async fn from_request_parts(parts: &mut Parts, _state: &S) -> Result<Self, Self::Rejection> {
         let claim = parts.extensions.get::<Claim>().ok_or(ErrorKind::Internal)?;
         let user_id = claim.sub.clone();
```
Comment (chesedo) [inline gateway/src/auth.rs:37]: Does it make sense to still have a `User` struct? We can just implement `FromRequestParts` for the claim right?

**Review by chesedo — COMMENTED** [review]
Small nit. If all tests are still passing then I don't see the need for updating any more tests.

File: gateway/src/auth.rs (hunk 1 above)
Comment (jonaro00) [reply gateway/src/auth.rs:37]: *only traits defined in the current crate can be implemented for a type parameter*
Claim can technically be moved to backends (it seems), but I'd skip it for now.

**Review by chesedo — APPROVED** [review]
LGTM

**Pull request merged**